src/tree_types.ml

```diff
@@ -39,14 +37,14 @@ module TCMap = Map.Make(TC)
 (** {3 Decision trees for rewriting} *)
 
 (** Representation of a branching conditions. *)
-type 'term tree_cond =
+type tree_cond =
   | CondNL of int * int
   (** Are the terms at the given indices convertible? We enforce the invariant
       that the first element is a point of reference, which appears in all the
       convertibility conditions involving a given non-linear variable. *)
-  | CondFV of 'term Bindlib.var array * int
-  (** Are the bound variables (which are free at the time of the checking) of
-      the term at the given index in the array? *)
+  | CondFV of int array * int
```

Comment:
Gabriel, would you mind to permute these arguments, that is, to take `CondFV of int * int array`?